lucet-module-data/src/signature.rs

```diff
@@ -76,25 +79,28 @@ impl RawModuleAndData {
         let mut obj_bin: Vec<u8> = Vec::new();
         File::open(&path)?.read_to_end(&mut obj_bin)?;
 
-        let module_data_symbol_data = Self::symbol_data(&obj_bin, "lucet_module_data", true)?
-            .ok_or(io::Error::new(
+        let native_data_symbol_data =
+            Self::symbol_data(&obj_bin, LUCET_MODULE_SYM, true)?.ok_or(io::Error::new(
                 io::ErrorKind::InvalidInput,
-                "module data not found",
+                format!("`{}` symbol not present", LUCET_MODULE_SYM),
             ))?;
-        let module_data_len_symbol_data =
-            Self::symbol_data(&obj_bin, "lucet_module_data_len", true)?.ok_or(io::Error::new(
+
+        // While `module_data` is the first field of the `NativeData` that `lucet_module` points
+        // to, it is a virtual address, not a file offset. The translation is somewhat tricky at
+        // the moment, so just look at the corresponding `lucet_module_data` symbol for now.
```

Comment:
Can we add a TODO here or in an issue to resolve this, if there's a path forward?

Reply:
https://github.com/fastly/lucet/issues/245 for the adventurous spirit